formula/src/main/java/com/instacart/formula/internal/SnapshotImpl.kt

```diff
@@ -91,15 +86,6 @@ internal class SnapshotImpl<out Input, State> internal constructor(
     }
 
     fun <Event> dispatch(transition: Transition<Input, State, Event>, event: Event) {
-        if (!running) {
-            throw IllegalStateException("Transitions are not allowed during evaluation")
-        }
-
-        if (!delegate.isTerminated() && delegate.isEvaluationNeeded(associatedEvaluationId)) {
```

Comment:
We don't need this check as transitions are controlled with a queue mechanism (lives in FormulaManager).